src/nvim/regexp.c

```diff
@@ -5043,7 +5043,7 @@ static bool regmatch(
             } else {
               MB_PTR_BACK(rex.line, rp->rs_un.regsave.rs_u.ptr);
               if (limit > 0
-                  && (long)(behind_pos.rs_u.ptr
+                  && (long)(behind_pos.rs_u.ptr  // NOLINT(google-readability-casting)
                             - rp->rs_un.regsave.rs_u.ptr) > limit) {
```

Comment:
perhaps we should cast limit to `ptrdiff_t` instead? (semantically preferred as `sizeof(ptr) >= sizeof(long)` )